x-pack/test/functional/services/machine_learning/job_wizard_common.ts

```diff
@@ -358,6 +388,39 @@ export function MachineLearningJobWizardCommonProvider({ getService }: FtrProvid
       await this.assertDateRangeSelection(expectedStartDate, expectedEndDate);
     },
 
+    async ensureAdditionalSettingsSectionOpen() {
+      await retry.tryForTime(5000, async () => {
+        if ((await testSubjects.exists(additionalSettingsSectionSelector())) === false) {
+          await testSubjects.click('mlJobWizardToggleAdditionalSettingsSection');
+          await testSubjects.existOrFail(additionalSettingsSectionSelector(), { timeout: 1000 });
+        }
+      });
+    },
+
+    async ensureNewCustomUrlFormModalOpen() {
+      await retry.tryForTime(5000, async () => {
+        if ((await testSubjects.exists(newCustomUrlFormModal())) === false) {
+          await testSubjects.click('mlJobOpenCustomUrlFormButton');
+          await testSubjects.existOrFail(newCustomUrlFormModal(), { timeout: 1000 });
+        }
+      });
+    },
+
+    async addCustomUrl() {
+      await this.ensureNewCustomUrlFormModalOpen();
+      // Fill-in the form
+      await customUrls.setCustomUrlLabel('check-kibana-dashboard');
```

Comment:
I would like to avoid setting static values in a service method. Instead, this should be a parameter and the test file can then call it with a test specific value. This helps to keep the service methods re-usable.

Reply:
Fixed